src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -833,6 +836,15 @@ FixedwingPositionControl::control_position(const Vector2f &curr_pos, const Vecto
 
 			float alt_sp = pos_sp_curr.alt;
 
+			if (pos_sp_next.type == position_setpoint_s::SETPOINT_TYPE_LAND && pos_sp_next.valid) {
+				// We're in a loiter directly before a landing WP. Enable our landing configuration (flaps,
```

Comment:
I don't think it is valid to assume that all FW landings are preceded by a loiter.

Reply:
See reply in my big comment.